internal/server/config.go

```diff
@@ -19,55 +18,106 @@ import (
 	"github.com/infrahq/infra/internal/server/data"
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/internal/server/providers"
+	"github.com/infrahq/infra/internal/validate"
 	"github.com/infrahq/infra/uid"
 )
 
 type Provider struct {
-	Name         string `validate:"required"`
-	URL          string `validate:"required"`
-	ClientID     string `validate:"required"`
-	ClientSecret string `validate:"required"`
+	Name         string
+	URL          string
+	ClientID     string
+	ClientSecret string
 	Kind         string
 	AuthURL      string
 	Scopes       []string
 }
 
+func (p Provider) ValidationRules() []validate.ValidationRule {
+	return []validate.ValidationRule{
+		// TODO: api.ValidateName
```

Comment:
Excited for this: consistent naming validation is so important for APIs

Reply:
I pushed another commit which resolves this TODO